### PR TITLE
Complete test case for kubeletconfig api validation

### LINKED ISSUE #55664: Need validate `EnforceNodeAllocatable` values in kubeletconfig api validation test.
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
> /kind bug
> /kind feature

**What happened**:
In kubeletconfig api validation test, it is incomplete about `EnforceNodeAllocatable`, see: 
https://github.com/kubernetes/kubernetes/blob/2d64ce5e8e45e26b02492d2b6c85e5ebfb1e4761/pkg/kubelet/apis/kubeletconfig/validation/validation_test.go#L29
and 
https://github.com/kubernetes/kubernetes/blob/2d64ce5e8e45e26b02492d2b6c85e5ebfb1e4761/pkg/kubelet/apis/kubeletconfig/validation/validation_test.go#L57
**What you expected to happen**:
Need validate all possible values for `EnforceNodeAllocatable`.
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by miaoyq**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #55664

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by miaoyq** [comment]
ping @yujuhong @mtaufen

File: pkg/kubelet/apis/kubeletconfig/validation/validation_test.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ func TestValidateKubeletConfiguration(t *testing.T) {
 
 	errorCase := &kubeletconfig.KubeletConfiguration{
 		CgroupsPerQOS:               false,
-		EnforceNodeAllocatable:      []string{"pods"},
+		EnforceNodeAllocatable:      []string{"pods", "system-reserved", "kube-reserved", "illegal-key"},
```
Comment (mtaufen) [inline pkg/kubelet/apis/kubeletconfig/validation/validation_test.go]: can we add system-reserved and kube-reserved to the success case too?

Comment (miaoyq) [reply pkg/kubelet/apis/kubeletconfig/validation/validation_test.go]: OK, will do :-)

Comment (miaoyq) [reply pkg/kubelet/apis/kubeletconfig/validation/validation_test.go]: Done.

**Pull request merged**